src/notebooks/controllers/controllerLoader.ts

```diff
@@ -125,16 +119,11 @@ export class ControllerLoader implements IControllerLoader, IExtensionSyncActiva
         }
     }
 
-    private async loadControllersImpl(cancelToken: vscode.CancellationToken) {
+    private async loadControllersImpl() {
         // First off set our context to false as we are about to refresh
```

Comment:
To be discussed with @IanMatthewHuff 
THe only thing that makes this async is updateing a context variable, which doesn't seem to be valid anymore.
I think we can update the context variable without relying on controlers,
instead we can directly use Python API to check if tehre are any interpreters and update the context var accordingly, and that can be done in a seprate class, making this class cleaner